src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -634,11 +634,13 @@ static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function>
         snippetsManager.register_pass<ngraph::snippets::pass::EnumerateNodes>();
         snippetsManager.register_pass<ngraph::snippets::pass::TokenizeSnippets>();
         snippetsManager.get_pass_config()->set_callback<ngraph::snippets::pass::TokenizeSnippets>(
-                [](const std::shared_ptr<const ov::Node>& n) -> bool {
+                [_tokenizeSoftmaxSnippets](const std::shared_ptr<const ov::Node>& n) -> bool {
                     // CPU Plugin support Swish in Subgraph via conversion to SwichCPU which assumes second input to be constant
                     if (ov::is_type<const ov::op::v4::Swish>(n)) {
                         if (n->inputs().size() > 1 && !ov::is_type<const ov::op::v0::Constant>(n->get_input_node_shared_ptr(1)))
                             return true;
+                    } else if (ov::is_type<const ov::op::v1::Softmax>(n) || ov::is_type<const ov::op::v8::Softmax>(n)) {
+                        return !_tokenizeSoftmaxSnippets;
```

Comment:
This will allow tokenization even if `has_only_const_inputs || bad_input_rank || bad_output_rank` is true. It looks like an undesired behavior (the same problem with Swish). Could you fix it please?